aten/src/ATen/core/function_schema.h

```diff
@@ -550,7 +550,10 @@ inline std::ostream& operator<<(std::ostream& out, const Argument& arg) {
   // in schema, we have Tensor?(a!) input, and t(a!)?.
   // however, t?(a!) doesn't work with schema parser.
   // so we always use Type(alias)? format
-  auto type = arg.type();
+  // real_type versus fake_type: in order to be compatible with FunctionSchema
+  // parser, printing an argument with either MemoryFormat or Layout type should
+  // give us the original schema string, hence printing out real_type.
+  auto type = arg.real_type();
```

Comment:
Would be good confirm with @ezyang. Since fake types were introduced in JIT we now have a divergence between schemas in JIT and schemas in torchgen, but it seems more important now for mobile to be able to stringify JIT schemas and expect to have parity with what you see in torchgen.

When using `torch.jit.script` or `torch.jit.trace`, are the actual schemas ever serialized (using this function)? If they are, then this might be BC breaking (and we might need to introduce a separate function for printing the "real" schema).

Reply:
The saving grace here is that, based on the tests here, it seems the string output of function schema is not actually used by anything substantive. So I'm OK with changing this over. It is technically BC breaking, but it's pretty irritating that the schema string you get from the C bindings is wrong. It might be a good idea to also provide the fake schema, but we can add that later.